src/flux_1/weights/weight_handler.py

```
@@ -6,21 +6,107 @@
+        lora_weights = tree_flatten(lora_transformer)
+        visited={}
+        
+        for key,weight in lora_weights:
+            splits=key.split(".")
+            target=transformer
+            visiting=[]
+            for splitKey in splits:
+                if isinstance(target,dict) and splitKey in target:
+                    target=target[splitKey]
+                    visiting.append(splitKey)
+                elif isinstance(target,list) and len(target)>0:
+                    if(len(target)< int(splitKey)):
+                        for _ in range(int(splitKey)-len(target)+1):
+                            target.append({})
+                        
+                    target=target[int(splitKey)]
+                    visiting.append(splitKey)
+                else:
+                    parentKey=".".join(visiting)
+                    if(parentKey in visited and 'lora_A' in visited[parentKey] and 'lora_B' in visited[parentKey]):
+                        continue
+                    if not splitKey.startswith("lora_"):
+                        visiting.append(splitKey)
+                        parentKey=".".join(visiting)
+                        if(splitKey=="net"):
+                            target['net']=list({})
+                            target=target['net']
+                        elif (splitKey=="0"):
+                            target.append({})
+                            target=target[0]
+                            continue
+                        elif (splitKey=="proj"):
+                            target[splitKey]=weight
+                            if parentKey not in visited:
+                                visited[parentKey]={}
+                        continue
+                    if parentKey not in visited:
+                        visited[parentKey]={}
+                    visited[parentKey][splitKey]=weight
+                    if not 'weight' in target:
+                        continue
```

The problem seems to be that some weights are not loaded at all. Notice that in the _transformer method some layers names are changed. So in the tree there are no matching keys for them and they get discarded with this `continue` statement. So we need to add some logic that changes the tree keys from huggingface to accommodate this.
More generally, if weights are ignored this should probably not fail silently. I would raise an error here every time it happens. The logic of descending the tree and matching weights should also probably be a bit more explicit, maybe through a separate class where all this is handled (as pointed out above by @awni)

---

Nice catch! I can say that this was also a common source of error when developing the project - thinking that weights were loaded when, in fact, they were not. Totally agree with @Xuzzo here, that we should be very explicit when weights are not being set. We should probably also have the same behaviour when loading the original weights just to be sure (in a future PR). Quickly digging around a bit, it might actually be better to use the `load_weights` method which has a `strict` option rather than what I currently did with `update`.  

Also, regarding the weight pre-processing to convert from Diffusers weights to MFLUX weights, while the transformations are not that big, it is a bit unfortunate to also have to think about them here too. 
For now, I think we could reuse that logic as @Xuzzo suggests if it is needed to make sure all LoRA weights are correctly set. Longer term, I can imagine that with some restructuring of the model setup, we can shorten/remove some of the weight transformations (i.e be more similar to the Diffusers layout), but this is more of a "nice to have" thing...